src/Data/List/NonEmpty.purs

```diff
@@ -132,16 +133,18 @@ singleton :: forall a. a -> NonEmptyList a
 singleton = NonEmptyList <<< NE.singleton
 
 cons :: forall a. a -> NonEmptyList a -> NonEmptyList a
-cons y (NonEmptyList (x :| xs)) = NonEmptyList (y :| x : xs)
+cons y (NonEmptyList (x :| xs)) = y :|| x : xs
 
 cons' :: forall a. a -> L.List a -> NonEmptyList a
-cons' x xs = NonEmptyList (x :| xs)
+cons' x xs = NonEmptyList (NonEmpty x xs)
+
+infixr 5 cons' as :||
 
 snoc :: forall a. NonEmptyList a -> a -> NonEmptyList a
-snoc (NonEmptyList (x :| xs)) y = NonEmptyList (x :| L.snoc xs y)
+snoc (NonEmptyList (x :| xs)) y = x :|| L.snoc xs y
```

Comment:
If we could get pattern matching to work, and reuse `:|`, we could rewrite this (and friends) as:
```purs
snoc (x :| xs) y = x :| L.snoc xs y
```